app/src/main/java/com/bashkevich/androidfundamentals/model/workmanager/MovieLoaderWorker.kt

```diff
@@ -0,0 +1,89 @@
+package com.bashkevich.androidfundamentals.model.workmanager
+
+import android.content.Context
+import android.util.Log
+import androidx.work.CoroutineWorker
+import androidx.work.WorkerParameters
+import com.bashkevich.androidfundamentals.model.database.MoviesDatabase
+import com.bashkevich.androidfundamentals.model.database.entity.MovieActorCrossRef
+import com.bashkevich.androidfundamentals.model.database.entity.MovieGenreCrossRef
+import com.bashkevich.androidfundamentals.model.network.RetrofitModule
+import com.bashkevich.androidfundamentals.model.repository.toActorEntity
+import com.bashkevich.androidfundamentals.model.repository.toGenreEntity
+import com.bashkevich.androidfundamentals.model.repository.toMovieEntity
+import com.bashkevich.androidfundamentals.moviesdetails.viewmodel.MoviesDetailsViewModel
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+class MovieLoaderWorker(appContext: Context, params: WorkerParameters) : CoroutineWorker(
+    appContext,
+    params
+) {
+    override suspend fun doWork(): Result = withContext(Dispatchers.IO) {
```

Comment:
Who is listening for the result?

Reply:
I think MainActivity

Reply:
My bad worker should return the result to know if it is should be rescheduled